src/world/action/jump.c

```diff
@@ -1,15 +1,250 @@
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6348_E24388, void);
+    phys_init_integrator_for_current_state();
+    collisionStatus = &gCollisionStatus;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B647C_E244BC, void);
+    if ((playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO)) {
+        temp_v1 = 0x90005;
+    }
+    else {
+        phi_a0 = (playerStatus->animFlags & PLAYER_ANIM_FLAG_3);
+        temp_v1 = 0x60009;
+        if (!phi_a0) {
+            temp_v1 = 0x10007;
+        }
+    }
+    suggest_player_anim_clearUnkFlag(temp_v1);
+    collisionStatus->lastTouchedFloor = collisionStatus->currentFloor;
+    collisionStatus->currentFloor = -1;
+}
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6508_E24548, void);
+void func_802B60B4_E240F4() {
+    s32 temp_v1;
+    s32 phi_a0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam;
+
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= 0x7FFFFFFF;
+        func_802B6000_E24040();
+
+        if (playerStatus->actionState == ACTION_STATE_LAUNCH) {
+            phys_adjust_cam_on_landing();
+        } else {
+            cam = &gCameras;
+            cam->moveFlags |= 1;
+        }
+
+        if (playerStatus->actionState == ACTION_STATE_JUMP) {
+            phi_a0 = SOUND_JUMP_2081;
+            if ((playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO)) {
+                phi_a0 = SOUND_JUMP_8BIT_MARIO;
+            }
+            sfx_play_sound_at_player(phi_a0, 0);
+        }
+    }
+
+    if ((playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO)) {
+        phi_a0 = 0x90005;
+    } else {
+        temp_v1 = playerStatus->animFlags & PLAYER_ANIM_FLAG_3;
+        phi_a0 = 0x60009;
+
+        if (!temp_v1) {
+          phi_a0 = 0x10007;
+        }
+    }
+    suggest_player_anim_clearUnkFlag(phi_a0);
+    playerStatus->decorationList++;
+}
+
+void func_802B6198_E241D8() {
+    f32 temp_f2;
+    s32 temp_s1;
+    s32 phi_a0;
+    Entity* temp_v0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    Camera* cam;
+
+    temp_s1 = 0x80000000;
+    if (playerStatus->flags < 0) {
+        temp_v0 = get_entity_by_index(collisionStatus->currentFloor);
+        temp_f2 = temp_v0->position.z;
+        D_8010C960 = temp_v0->position.x;
+        D_8010C97C = temp_f2;
+        func_802B6000_E24040();
+        playerStatus->flags |= 0x880000;
+        disable_player_input();
+    }
+
+    playerStatus->decorationList++;
+    if (playerStatus->fallState == 1) {
+        if ((playerStatus->flags & 0x80000000)) {
+
+          playerStatus->flags &= 0x7FFFFFF5;
+          playerStatus->flags |= 4;
+
+          phi_a0 = 0x6000A;
+
+          if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_3)) {
+            phi_a0 = 0x10008;
+          }
+
+          temp_s1 = phi_a0;
+          suggest_player_anim_clearUnkFlag(temp_s1);
+          cam = &gCameras;
+          cam->moveFlags |= 1;
```

Comment:
Are you sure this camera temp is needed? You should be able to do `gCameras[0].moveFlags`. If that doesn't work, the value of cam should actually be `&gCameras[0]` since it's a pointer to the first camera in the array, rather than a pointer to the array itself.